ocpp-common/src/main/java/eu/chargetime/ocpp/model/OcppRequest.java

```diff
@@ -15,6 +15,13 @@ public abstract class OcppRequest implements Request {
 	@Override
 	public void setRawMessageJson(String rawMessageJson){
 		this.rawMessageJson = rawMessageJson;
+
+		// align setting and getting of rawMessageJson field with the requestId field
+		if(rawMessageJson == null){
+			requestId = null;
+		} else {
+			setRequestIdFromRawMessageJson();
```

Comment:
this is not needed. It will be recalculated on the next getRequestId() anyways. All this PR needs to do is add a requestId = null; to setRawMessageJson. That means it will be recalculated next time getRequestId is called which will return null if the rawJson was null

Reply:
Done :)